actioncable.go

```diff
@@ -94,6 +97,44 @@ func (c *Client) Subscribe(channel string) (<-chan *EventOrErr, error) {
 	}
 
 	c.subscriptions[channel] = make(chan *EventOrErr, 32)
+	c.subscData[channel], err = json.Marshal(&innerIdentifier{Channel: channel})
+	if err != nil {
+		panic(err)
+	}
+
+	// Add subscription command to outbound queue.
+	go func() {
+		c.subc <- channel
+	}()
+
+	return c.subscriptions[channel], nil
+}
+
+var ErrNoChannelName = errors.New("Subscription data must contain a channel name")
+
+// SubscribeWith subscribes to a channel giving parameters
+// the params must contain a field "channel" containing the channe to subscribe to
+func (c *Client) SubscribeWith(params json.RawMessage) (<-chan *EventOrErr, error) {
```

Comment:
A couple of things feel weird about this API to me:
* I don't recall seeing other Go APIs that take a `json.RawMessage` as an input argument. Doesn't mean it wouldn't ever make sense, but just feels weird.
* I'm not super happy with the fact that params has to be unmarshaled anyway, presumably right after the caller finished marshaling it.

I think a more idiomatic API might be:

```go
func (c *Client) SubscribeWith(channel string, params map[string]interface{}) (<-chan *EventOrErr, error) {
  ...
}
```

While I dislike `map[string]interface{}` for a number of reasons, I'm not sure there's a better alternative for this scenario where you want to let folks pass in arbitrary key-value data that you can also merge into (with `"channel": "channelName"`).

This would have a couple of advantages I can think of:

* channel name is still explicit
* no need for an `ErrNoChannelName`
* ability to DRY up `Subscribe`/`SubscribeWith` by just making `Subscribe` call `SubscribeWith` using a `nil` map (the latter just needs to handle a `nil` map)
* After setting `"channel"` on the params map, you can serialize it just once here and still save it as a `json.RawMessage` (or maybe just `[]bytes`?) on the `Client`. That way there is no redundant serialization or deserialization happening inside or outside this function.